### PR TITLE
Implement flags grouping for karmada-scheduler

### LINKED ISSUE #1485: [umbrella] Grouping flags for components
**What would you like to be added**:
We have implemented to group flags for `karmada-controller-manager` at #1468  and `karmada-agent` at #1389 , and the other components also need to be implemented.

- [x] karmada-aggregated-apiserver (#1494 by @AllenZMC )
- [x] karmada-scheduler (#1491 by @jameszhangyukun )
- [x] karmada-descheduler (#1490 by @AllenZMC )
- [x] karmada-scheduler-estimator (#1493 by @AllenZMC )
- [x] karmada-webhook (#1488 by @AllenZMC )

**Why is this needed**:

### PR DESCRIPTION
**Pull request by jameszhangyukun**
Signed-off-by: zhangyukun <38148677+jameszhangyukun@users.noreply.github.com>

**What type of PR is this?**
/kind feature

**What this PR does / why we need it**:
Implement flags grouping for karmada-scheduler
**Which issue(s) this PR fixes**:
Part of #1485

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note

```

### CODE REVIEW

**Comment by lonelyCZ** [comment]
Please complete note-release.

File: cmd/scheduler/app/scheduler.go [hunk 1]
```diff
@@ -54,13 +57,22 @@ func NewSchedulerCommand(stopChan <-chan struct{}) *cobra.Command {
...
 	}
 
-	// Init log flags
-	// TODO(@RainbowMango): Group the flags to "logs" flag set.
-	klog.InitFlags(flag.CommandLine)
+	fss := cliflag.NamedFlagSets{}
 
-	opts.AddFlags(cmd.Flags())
+	genericFlagSet := fss.FlagSet("generic")
+	// genericFlagSet.AddGoFlagSet(flag.CommandLine)
```
Comment (lonelyCZ) [inline cmd/scheduler/app/scheduler.go:63]: We could delete this comment.

**Comment by lonelyCZ** [comment]
Could you please change `Fixes #1485` to `Part of #1485`? @jameszhangyukun 

```
What type of PR is this?
/kind feature

What this PR does / why we need it:
Implement flags grouping for karmada-scheduler
Which issue(s) this PR fixes:
Fixes https://github.com/karmada-io/karmada/issues/1485
```

**Comment by RainbowMango** [comment]
Yes, the `Fixes #1485` instruction will close the issue.

**Comment by jameszhangyukun** [comment]
> Could you please change `Fixes #1485` to `Part of #1485`? @jameszhangyukun
> 
> ```
> What type of PR is this?
> /kind feature
> 
> What this PR does / why we need it:
> Implement flags grouping for karmada-scheduler
> Which issue(s) this PR fixes:
> Fixes https://github.com/karmada-io/karmada/issues/1485
> ```

i fix it.

**Review by RainbowMango — APPROVED** [review]
/approve
Thanks.

**Pull request merged**